spec/services/exchange_rates/update_currency_rates_service_spec.rb

```diff
@@ -1,59 +1,91 @@
 require 'rails_helper'
 
 RSpec.describe ExchangeRates::UpdateCurrencyRatesService do
+  let(:xe_api) { instance_double(ExchangeRates::XeApi) }
+
+  let(:response) do
+    {
+      'to' => [
+        { 'quotecurrency' => 'AED', 'mid' => 4.662353708 }, # Has corresponding currency
+        { 'quotecurrency' => 'AED', 'mid' => 4.662353708 }, # Has corresponding currency but is duplicated and won't create two
+        { 'quotecurrency' => 'EUR', 'mid' => 6.662353708 }, # Has a corresponding currency
+        { 'quotecurrency' => 'FGA', 'mid' => 5.662353708 }, # Has no corresponding currency
+      ],
+    }
+  end
+
+  before do
+    allow(ExchangeRates::XeApi).to receive(:new).with(date: sample_date).and_return(xe_api)
+    allow(xe_api).to receive(:get_all_historic_rates).and_return(response)
+  end
+
   describe '#call' do
-    subject(:service) { described_class.new(date, sample_date) }
-
-    let(:date) { Time.zone.today.next_month }
-    let(:expected_rates) do
-      [
-        {
-          currency_code: 'AED',
-          validity_start_date: date.beginning_of_month,
-          validity_end_date: date.end_of_month,
-          rate: 4.662353708,
-          rate_type: ExchangeRateCurrencyRate::MONTHLY_RATE_TYPE,
-        },
-        {
-          currency_code: 'EUR',
-          validity_start_date: date.beginning_of_month,
-          validity_end_date: date.end_of_month,
-          rate: 6.662353708,
-          rate_type: ExchangeRateCurrencyRate::MONTHLY_RATE_TYPE,
-        },
-      ].as_json
-    end
-    let(:sample_date) { Time.zone.today }
-
-    let(:xe_api) { instance_double(ExchangeRates::XeApi) }
-    let(:response) do
-      {
-        'to' => [
-          { 'quotecurrency' => 'AED', 'mid' => 4.662353708 }, # Has corresponding currency
-          { 'quotecurrency' => 'AED', 'mid' => 4.662353708 }, # Has corresponding currency but is duplicated and won't create two
-          { 'quotecurrency' => 'EUR', 'mid' => 6.662353708 }, # Has a corresponding currency
-          { 'quotecurrency' => 'FGA', 'mid' => 5.662353708 }, # Has no corresponding currency
-        ],
-      }
-    end
+    context 'when type is MONTHLY_RATE_TYPE' do
+      subject(:service) { described_class.new(date, sample_date, 'monthly') }
 
-    before do
-      allow(ExchangeRates::XeApi).to receive(:new).with(date: sample_date).and_return(xe_api)
-      allow(xe_api).to receive(:get_all_historic_rates).and_return(response)
-      create(:exchange_rate_country_currency, currency_code: 'AED')
-      create(:exchange_rate_country_currency, currency_code: 'EUR')
-    end
+      let(:date) { Time.zone.today.next_month }
+      let(:expected_rates) do
+        [
+          {
+            currency_code: 'AED',
+            validity_start_date: date.beginning_of_month,
+            validity_end_date: date.end_of_month,
+            rate: 4.662353708,
+            rate_type: ExchangeRateCurrencyRate::MONTHLY_RATE_TYPE,
```

Comment:
As you use the constant here maybe set this up as a let that you can use around the code as well as where you use 'monthly'

Reply:
it's only used in one place and I think it would just make the code look less readable

Reply:
Its used in 3 places?